agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -137,5 +147,30 @@ def sync_state_to_master(current_state)
       rpc_client.async.notification('/node_service_pods/set_state', [node.id, data])
       @prev_state = current_state
     end
+
+    # @param [Docker::Container] service_container
+    def ensure_health_check(service_container)
+      return if service_container.nil? || service_container.labels['io.kontena.health_check.uri'].nil?
+      return if @health_check && @health_check.alive?
+
+      @health_check = ContainerHealthCheckWorker.new(service_container)
+      @health_check.async.start
+    rescue => exc
+      error "#{exc.class.name}: #{exc.message}"
+    end
+
+    def stop_health_check
+      if @health_check
+        @health_check.terminate if @health_check.alive?
+      end
+    end
+
+    private
+
+    def finalize
```

Comment:
Oh right, this is a Celluloid finalizer, not a Ruby finalizer: https://github.com/celluloid/celluloid/wiki/Finalizers

Stil... linking instead?